components/combobox/metadata/combobox.yml

```
@@ -0,0 +1,279 @@
+name: Combobox
+description: Comboboxes combine a text field with a picker menu.
+SpectrumSiteSlug: https://spectrum.adobe.com/page/combo-box/
+sections:
+  - name: Migration Guide
+    description: |
+      ### Component separated from InputGroup
+      This component was previously within InputGroup. It is now separated out into its own component (v1.0.0).
+      The classes containing `InputGroup` have been renamed or removed. For details, see the "renamed" and "removed" sections below or the example markup.
+
+      ### New Picker markup
+      Instead of a `.spectrum-Picker`, Combobox now uses `.spectrum-PickerButton`. Refer to the example markup for usage details.
+
+      ### New Textfield markup
+      Combobox now uses the new Textfield markup. See the [Textfield migration guide](textfield.html#migrationguide) for more information.
+
+      ### Additional clases
+      The following classes must be added:
+
+      * `.spectrum-Combobox-textfield` is now required on the Textfield outer element (`.spectrum-Textfield`)
+      * `.spectrum-Combobox-input` is now required on the `<input>` element inside of Textfields (`.spectrum-Textfield-input`)
+      * `.spectrum-Combobox-button` is now required on the FieldButton (`.spectrum-ActionButton spectrum-ActionButton--sizeM`)
+
+      ### Indicating validity and focus
+      Validity and focus must be bubbled up to the parent so descendants siblings can be styled.
+
+      Thus, implementations must add the following classes in the following situations:
+
+      * `.spectrum-Combobox.is-focused` - when the input or button is focused with the mouse
+      * `.spectrum-Combobox.is-keyboardFocused` - when the input or button is focused with the keyboard
+      * `.spectrum-Combobox.is-valid` - when the input has an explicit valid state
+      * `.spectrum-Combobox.is-invalid` - when the input has an explicit invalid state
+      * `.spectrum-Combobox.is-disabled` - when the control is disabled. You must also add `.is-disabled` to the `.spectrum-Combobox-textfield` and `[disabled]` to the `.spectrum-Combobox-button`.
+      * `.spectrum-Combobox.is-loading` - when the progress circle is being shown
```

Since each of these are applied to the parent (unless explicitly explained in the description), should we simplify the doc by noting the class once at the top?

```suggestion
      Thus, implementations should add the following classes to the `.spectrum-Combobox` parent class in the following situations:

      * `.is-focused` - when the input or button is focused with the mouse
      * `.is-keyboardFocused` - when the input or button is focused with the keyboard
      * `.is-valid` - when the input has an explicit valid state
      * `.is-invalid` - when the input has an explicit invalid state
      * `.is-disabled` - when the control is disabled; should also add to the `.spectrum-Combobox-textfield` and include a `[disabled]` attribute to the `.spectrum-Combobox-button`
      * `.is-loading` - when the progress circle is being shown
```

---

Good idea. Updated in commit [9e728ef](https://github.com/adobe/spectrum-css/pull/1556/commits/9e728ef7e359d728fc3d0f2efebc4a3d1e7b84c5)